cfg/config.go

```diff
@@ -1,60 +1,150 @@
 package cfg
 
 import (
-	"encoding/json"
-	"io/ioutil"
+	"fmt"
+	"net/url"
 	"os"
+	"path/filepath"
 
 	"github.com/muesli/beehive/bees"
+	gap "github.com/muesli/go-app-paths"
+	log "github.com/sirupsen/logrus"
 )
 
+const appName = "beehive"
+
+var cfgFileName = "beehive.conf"
+
 // Config contains an entire configuration set for Beehive
 type Config struct {
 	Bees    []bees.BeeConfig
 	Actions []bees.Action
 	Chains  []bees.Chain
+	backend IConfigBackend
+	url     *url.URL
+}
+
+// IConfig is the interface implemented by the configuration backends
```

Comment:
I must admit I chuckled a little when I saw `IConfig`. Let's try and keep using a regular name here, as idiomatic Go suggests. Also, the doc-comment doesn't match the declared type below: `IConfig` vs `IConfigBackend`.

We could call this just `ConfigBackend` or may be something like `ConfigHandler`.

Reply:
Fixed